apollo-router/src/plugins/telemetry/config_new/conditions.rs

```diff
@@ -55,28 +56,95 @@ impl<T> Condition<T>
 where
     T: Selector,
 {
-    pub(crate) fn evaluate_request(&mut self, request: &T::Request) -> Option<bool> {
+    /// restricted_stage is Some if this condiiton will only applies at a specific stage like for events for example
+    pub(crate) fn validate(&self, restricted_stage: Option<Stage>) -> Result<(), String> {
```

Comment:
Let's talk about this. This is to catch selectors that will be triggered for stages after the containing event has taken place right?

Reply:
Exactly